File: JS-Basics/05-Conditional-Statements-Advanced-Lab/p12_Trade_Commissions.js
function tradeCommissions(input) {
    let city = input[0];
    let number = Number(input[1]);
    let bonus = 0;

    switch (city) {
        case 'Sofia':
            if (number >= 0 && number <= 500) {
                bonus = number * 0.05;
            } else if (number > 500 && number <= 1000) {
                bonus = number * 0.07;

            } else if (number > 1000 && number <= 10000) {
                bonus = number * 0.08;

            } else {
                bonus = number * 0.12;
            }
            break;
        case 'Varna':
            if (number >= 0 && number <= 500) {
                bonus = number * 0.045;
            } else if (number > 500 && number <= 1000) {
                bonus = number * 0.075;

            } else if (number > 1000 && number <= 10000) {
                bonus = number * 0.1;

            } else {
                bonus = number * 0.13;
            }
            break;
        case 'Plovdiv':
            if (number >= 0 && number <= 500) {
                bonus = number * 0.055;
            } else if (number > 500 && number <= 1000) {
                bonus = number * 0.08;

            } else if (number > 1000 && number <= 10000) {
                bonus = number * 0.12;

            } else {
                bonus = number * 0.145;
            }
            break;
        default:
            break;
    }

    if (bonus <= 0) {
        console.log('error');
    } else {
        console.log(bonus.toFixed(2));
    }
}

tradeCommissions(["Sofia", "1500"]);
tradeCommissions(["Plovdiv", "499.99"]);
tradeCommissions(["Varna", "3874.50"]);
tradeCommissions(["Kaspichan", "-50"]);

File: Programming-Basic/05-Conditional-Statements-Advanced-Lab/p12_Trade_Commissions.java
import java.util.Scanner;

public class p12_Trade_Commissions {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String city = scanner.nextLine();
        double sales = Double.parseDouble(scanner.nextLine());
        double commission = 0.0;

// Град	0 ≤ s ≤ 500	500 < s ≤ 1 000	1 000 < s ≤ 10 000	s > 10 000
//Sofia	5%	7%	8%	12%

        switch (city) {
            case "Sofia":
                if (sales >= 0 && sales <= 500) {
                    commission = sales * 0.05;
                } else if (sales > 500 && sales <= 1000) {
                    commission = sales * 0.07;
                } else if (sales > 1000 && sales <= 10000) {
                    commission = sales * 0.08;
                } else if (sales > 10000) {
                    commission = sales * 0.12;
                }
                break;
//Varna	4.5%	7.5%	10%	13%

            case "Varna":
                if (sales >= 0 && sales <= 500) {
                    commission = sales * 0.045;
                } else if (sales > 500 && sales <= 1000) {
                    commission = sales * 0.075;
                } else if (sales > 1000 && sales <= 10000) {
                    commission = sales * 0.10;
                } else if (sales > 10000) {
                    commission = sales * 0.13;
                }
                break;
//Plovdiv	5.5%	8%	12%	14.5%
            case "Plovdiv":
                if (sales >= 0 && sales <= 500) {
                    commission = sales * 0.055;

                } else if (sales > 500 && sales <= 1000) {

                    commission = sales * 0.08;
                } else if (sales > 1000 && sales <= 10000) {

                    commission = sales * 0.12;
                } else if (sales > 10000) {

                    commission = sales * 0.145;
                }
                break;
        }
        if (commission == 0.0) {
            System.out.println("error");
        } else {
            System.out.printf("%.2f", commission);
        }
    }
}
